App shortcuts sample

gonzalonm opened the pull request:
Sample ported from https://github.com/googlesamples/android-AppShortcuts

pjcollins commented on android-n/AppShortcuts/AppShortcuts/Main.cs:
```diff
@@ -0,0 +1,281 @@
+			public MyAdapter(Context context)
+			{
+				Context = context;
+				Inflater = (LayoutInflater)Context.GetSystemService(Java.Lang.Class.FromType(typeof(LayoutInflater)));
```
Can be `GetSystemService(Context.LayoutInflaterService)`

gonzalonm replied:
Done

pjcollins commented on android-n/AppShortcuts/AppShortcuts/Main.cs:
```diff
@@ -0,0 +1,281 @@
+				.SetView(editUri)
+				.SetPositiveButton("Add", (dialog, whichButton) =>
+				{
+					var url = editUri.Text.ToString().Trim();
```
Don't need a ToString () call on the Text property.

gonzalonm replied:
Done

pjcollins commented on android-n/AppShortcuts/AppShortcuts/ShortcutHelper.cs:
```diff
@@ -0,0 +1,274 @@
+		public ShortcutHelper(Context context)
+		{
+			mContext = context;
+			mShortcutManager = (ShortcutManager)mContext.GetSystemService(Java.Lang.Class.FromType(typeof(ShortcutManager)));
```
Can use `GetSystemService (Context.ShortcutService)` here.

pjcollins commented:
Hey I noticed one issue with the port - the application crashes when you try to add too many app shortcuts (more then 5 I think?). This may just be a case of not catching the right exception type? The java version displays a toast with exception information in this case.
Stacktrace:
https://gist.github.com/pjcollins/f9358d1178235ed2a4e9d028dd8c6472

Other than that, there's a few minor stylistic changes
* We should remove the private keyword in variable declarations / method signatures which are private by default.
*  I've provided a couple inline comments for minor readability changes.

gonzalonm commented:
Regarding with the app crashes when you try to add too many shortcuts, I've added a toast with the exception information.

Also, I've removed the private keyboard from variable declarations and method signatures.

pjcollins commented:
WRT the try/catch for the Shortcut Manager invocation, I took a closer look at how the java lambdas were being used in the upstream sample and we can better emulate that to handle this scenario, please see the diff attached below:

https://gist.github.com/pjcollins/179c1e67a7d8e39189349408976f044b

gonzalonm commented:
Changes applied

pjcollins commented:
Looks good!
